w3_d3_2_update_and_delete/README.md

```diff
@@ -0,0 +1,202 @@
+  </form>
+</div>
+```
+
+```js
+// client-side JavaScript
+
+// remember:
+var phrasesController = {}
+phrasesController.template = _.template($('#phrase-template').html());
+
+phrasesController.update = function(phraseId, updatedWord, updatedDefinition) {
+  // send PUT request to server to update phrase
+  $.ajax({
+    type: 'PUT',
+    url: '/phrases/' + phraseId,
+    data: {
+      word: updatedWord,
+      definition: updatedDefinition
+    },
+    success: function(data) {
+      // pass phrase object through template and append to view
+      var $phraseHtml = $(phrasesController.template(data));
+      $('#phrase-' + phraseId).replaceWith($phraseHtml);
+    }
+  });
+};
+```
+
+```js
+// server-side JavaScript
+
+// remember:
+var phrases = [
+  {id: 0, word: 'REPL', definition: 'Read, Eval, Print, Loop'}
+];
+
+app.put('/phrases/:id', function(req, res) {
+
+  // set the value of the id
+  var targetId = parseInt(req.params.id);
+
+  // find item in `phrases` array matching the id
+  var foundPhrase = _.findWhere(phrases, {id: targetId});
+
+  // if form gave us a new word, update the phrase's word
+  foundPhrase.word = req.body.word || foundPhrase.word;
+
+  // if form gave us a new definition, update that
+  foundPhrase.definition = req.body.definition || foundPhrase.definition;
+
+  // send back edited object
+  res.json(foundPhrase);
+});
+```
+
+### Delete (aka DESTROY)
+
+```html
+<!-- delete button -->
+<button class="close right-side delete-phrase" aria-label="Close">
+  <span aria-hidden="true">&times;</span>
+</button>
+```
+
+```js
+// client-side JavaScript
+
+// remember:
+var phrasesController = {}
+
+phrasesController.delete = function(phraseId) {
+  // send DELETE request to server to delete phrase
+  $.ajax({
+    type: 'DELETE',
+    url: '/phrases/' + phraseId,
+    success: function(data) {
+      // remove deleted phrase li from the view
+      $('#phrase-' + phraseId).remove();
+    }
+  });
+}
+```
+
+```js
+// server-side JavaScript
+
+// remember:
+var phrases = [
+  {id: 0, word: 'REPL', definition: 'Read, Eval, Print, Loop'}
+];
+
+app.delete('/phrases/:id', function(req, res) {
+
+  // set the value of the id
+  var targetId = parseInt(req.params.id);
+
+  // find item in `phrases` array matching the id
+  var foundPhrase = _.findWhere(phrases, {id: targetId});
+
+  // get the index of the found item
+  var index = phrases.indexOf(foundPhrase);
+
+  // remove the item at that index, only remove 1 item
+  phrases.splice(index, 1);
+
+  // send back deleted object
+  res.json(foundPhrase);
+});
+```
+
+## Challenges (& Tonight's Homework)
+
+Implement the "D" (delete) in CRUD in the RESTful API you started this morning.
+
+### Requirements
+
+1. A user should be able to click a button on the page to indicate they'd like to delete an item.
+2. Your client-side JavaScript should register clicks on your delete button and send an DELETE request (via AJAX) to the server.
```

Comment:
should be "a" DELETE request right? I would fix this myself, but can't remember how to do it! Next time :)